src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -37,12 +47,29 @@ public class InstallToolCommand : CommandBase
             _configFilePath = appliedCommand.ValueOrDefault<string>("configfile");
             _framework = appliedCommand.ValueOrDefault<string>("framework");
             _source = appliedCommand.ValueOrDefault<string>("source");
+            _executablePackagePath = new DirectoryPath(new CliFolderPathCalculator().ExecutablePackagesPath);
+
+            _toolPackageObtainer = toolPackageObtainer ?? new ToolPackageObtainer(
+                                       _executablePackagePath,
+                                       () => new DirectoryPath(Path.GetTempPath())
+                                           .WithSubDirectories(Path.GetRandomFileName())
+                                           .WithFile(Path.GetRandomFileName() + ".csproj"),
+                                       new Lazy<string>(BundledTargetFramework.GetTargetFrameworkMoniker),
+                                       new PackageToProjectFileAdder(),
+                                       new ProjectRestorer());
+
+            _environmentPathInstruction = environmentPathInstruction
+                                          ?? EnvironmentPathFactory
+                                              .CreateEnvironmentPathInstruction();
+
+            _shellShimMaker = shellShimMaker ?? new ShellShimMaker(_executablePackagePath.Value);
+
+            _reporter = reporter ?? Reporter.Output;
         }
 
         public override int Execute()
         {
             var executablePackagePath = new DirectoryPath(new CliFolderPathCalculator().ExecutablePackagesPath);
```

Comment:
Do we still need this local vs. `_executablePackagePath` and, subsequently, the argument to `ObtainPackage`?

Reply:
you are right, should use _executablePackagePath  instead

Reply:
Seems like `_executablePackagePath` could be made a local to the constructor, though.

Reply:
Fixed, change _executablePackagePath  local